modules/oracle-xe/src/main/java/org/rnorth/testcontainers/containers/OracleContainer.java

```diff
@@ -0,0 +1,90 @@
+package org.rnorth.testcontainers.containers;
+
+import com.spotify.docker.client.messages.ContainerConfig;
+import com.spotify.docker.client.messages.ContainerInfo;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import com.zaxxer.hikari.pool.PoolInitializationException;
+
+/**
+ * Created by gusohal on 26/04/15.
+ */
+public class OracleContainer extends AbstractContainer implements DatabaseContainer  {
+
+    private static final String IMAGE = "alexeiled/docker-oracle-xe-11g";
+    private String oraclePort;
+
+    @Override
+    protected void containerIsStarting(ContainerInfo containerInfo) {
+        oraclePort = containerInfo.networkSettings().ports().get("1521/tcp").get(0).hostPort();
+    }
+
+    @Override
+    protected String getLivenessCheckPort() {
+        return oraclePort;
+    }
+
+    @Override
+    protected ContainerConfig getContainerConfig() {
+        return ContainerConfig.builder()
+                .image(getDockerImageName())
+                .exposedPorts("1521", "22", "8080")
+                .build();
+    }
+
+    @Override
+    protected String getDockerImageName() {
+        return IMAGE + ":" + tag;
+    }
+
+    @Override
+    public String getName() {
+        return "oracle";
+    }
+
+    @Override
+    public String getDriverClassName() {
+        return "oracle.jdbc.OracleDriver";
+    }
+
+    @Override
+    public String getJdbcUrl() {
+        return "jdbc:oracle:thin:" + getUsername() + "/" + getPassword() + "@//" + dockerHostIpAddress + ":" + oraclePort + "/xe";
+    }
+
+    @Override
+    public String getUsername() {
+        return "system";
+    }
+
+    @Override
+    public String getPassword() {
+        return "oracle";
+    }
+
+    @Override
+    public void waitUntilContainerStarted() {
+        super.waitUntilContainerStarted();
+        HikariConfig hikariConfig = new HikariConfig();
+        hikariConfig.setJdbcUrl(getJdbcUrl());
+        hikariConfig.setConnectionTestQuery("SELECT 1 FROM dual");
+        hikariConfig.setMinimumIdle(1);
+        hikariConfig.setMaximumPoolSize(1);
+
+        for (int i = 0; i < 30000; i++) {
```

Comment:
I think this could end up waiting for a very long time if the container fails to start (30000 * 5s)...

It looks like in the default AbstractContainer method it's waiting (6000 * 0.1s) and even that is, on reflection, a mistake on my part (should probably be 600 * 0.1s).

In the context this is to be used, can we reduce the max wait time to a minute or so?